app/models/contract.rb

```diff
@@ -296,7 +296,7 @@ def change_plan_internal(new_plan, &block)
   private
 
   def reset_counter_cache_for
-    [:plan].freeze
+    [:plan, :account].freeze
```

Comment:
First, RuboMarta strikes 😝 :
```suggestion
    %i[plan account].freeze
```
Now my "questions/suggestions":
1. Is it `account` or `user_account` ? why?
2. All this needs a test 😄 
3. Shouldn't the next method (`update_counter_cache?`) have this same association in the first `when` ? if not, why? because I believe that it shouldn't update this cache if the account is being deleted 😕

Reply:
@macejmic

Reply:
`alias account user_account` so it does not matter, if it's `account` or `user_account`.

Reply:
3. https://github.com/3scale/porta/blob/master/app/models/contract.rb#L302

so the counter is not being updated when the provider account is being deleted.

Reply:
2. this is an "integration" unit test that covers the main use case:

```
test 'bought_cinstances_count when deleting a cinstance' do
    account = FactoryBot.create(:simple_provider, provider_account: master_account)
    cinstance = account.buy!(master_account.default_service.published_plans.first!)

    assert_difference -> { account.reload.bought_cinstances_count }, -1 do
      cinstance.destroy
      account.reload.bought_cinstances_count
    end
  end
```

all other are already covered..